packages/core/src/lib/passcode.ts

```diff
@@ -0,0 +1,30 @@
+import { PasscodeType } from '@logto/schemas';
+import { customAlphabet, nanoid } from 'nanoid';
+
+import {
+  deletePasscodesByIds,
+  findUnconsumedPasscodesBySessionIdAndType,
+  insertPasscode,
+} from '@/queries/passcode';
+
+const randomCode = customAlphabet('1234567890', 6);
+
+export const createPasscode = async (
+  sessionId: string,
+  type: PasscodeType,
+  payload: { phone: string } | { email: string }
+) => {
+  // Disable existing passcodes.
+  const passcodes = await findUnconsumedPasscodesBySessionIdAndType(sessionId, type);
```

Comment:
nit: not 100% sure, but, do you think we could make the query method more flexible? Like 

```
findPasscodes({
 sessionId,
 type,
 consumed: false
})
```